charts/mirrorbits-parent/templates/ingress.yaml

```diff
@@ -0,0 +1,59 @@
+{{- if .Values.ingress.enabled -}}
+{{- $fullName := include "mirrorbits-parent.name" . -}}
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: {{ $fullName }}
+  labels:
+{{ include "mirrorbits-parents.labels" . | indent 4 }}
+  {{- with .Values.ingress.annotations }}
+  annotations:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+spec:
+{{- if .Values.ingress.className }}
+  ingressClassName: {{ .Values.ingress.className }}
+{{- end }}
+{{- if .Values.ingress.tls }}
+  tls:
+  {{- range .Values.ingress.tls }}
+    - hosts:
+      {{- range .hosts }}
+        - {{ . | quote }}
+      {{- end }}
+      secretName: {{ .secretName }}
+  {{- end }}
+{{- end }}
+  rules:
+  {{- range .Values.ingress.hosts }}
+    - host: {{ .host | quote }}
+      http:
+        paths:
+        {{- range .paths }}
+          - path: {{ .path }}
+            pathType: {{ .pathType | default "Prefix" }}
+            backend:
+            {{- if eq .backendService "mirrorbits" }}
+              {{- if not (index $.Values "mirrorbits-lite" "enabled") }}
+                {{- fail "Cannot use mirrorbits(-lite) as backend if it is disabled." }}
+              {{- end }}
+              service:
+                # TODO: update child chart to use a global value to let the parent-chart defined/get the value
+                name: {{ $.Release.Name | trunc 63 | trimSuffix "-" -}}-mirrorbits-lite
```

Comment:
All these names are risky: let say someone puts a really long release name like [Rindfleischetikettierungsuberwachungsaufgabenubertragungsgesetz](https://fr.wikipedia.org/wiki/Rindfleisch%C2%ADetikettierungs%C2%AD%C3%BCberwachungs%C2%ADaufgaben%C2%AD%C3%BCbertragungs%C2%ADgesetz), what happens then? (Spoiler: it will fail)

Reply:
You are correct. The naming here only relies on the subchart templating which is responsible for the size of the generated name through their template function.

I've deferred the resolution of this challenge as per the TODO comment. If we can pass an overiding name (even a long one) to the child chart, there should be no problem at all as the subchart templating function would truncate it (you already added smart logic in it already to deduplicate the release name btw.)

But it's not the role of the parent chart as per the encapsulation principle.